### PR TITLE
Upgrade firecracker-containerd binaries with containerd 1.5.2

### PR DESCRIPTION
**Pull request by plamenmpetrov**
- [x] upgrade rootfs and runtime binary to v0.24 
- [x] added artifact to integ test 
- [x] change format of vmID 
- [x] update firecracker-containerd binaries
- [x] clean up shim sockets 
- [x] cleaned go.sum 
- [x] add upgrade to CHANGELOG.md 

Signed-off-by: Plamen Petrov <plamb0brt@gmail.com>

### CODE REVIEW

**Review by ustiugov — CHANGES_REQUESTED** [review]
LGTM but let's pull the latest changes for firecracker-containerd. particularly, we are interested in containerd 1.5

**Review by ustiugov — COMMENTED** [review]
do we have to fix the grpc version? why?

File: go.mod [hunk 1]
```diff
@@ -2,8 +2,13 @@ module github.com/ease-lab/vhive
 
 go 1.15
 
-// Workaround for github.com/containerd/containerd issue #3031
-replace github.com/docker/distribution v2.7.1+incompatible => github.com/docker/distribution v2.7.1-0.20190205005809-0d3efadf0154+incompatible
+replace (
+	// Pin gPRC-related dependencies as like containerd v1.5.1
```
[the remark this answers is not in the document]
Comment (plamenmpetrov) [reply go.mod:6]: Might be. Do you want to fix it?

File: go.mod [hunk 2]
```diff
@@ -36,38 +41,33 @@ replace (
 
 replace (
 	github.com/ease-lab/vhive/examples/protobuf/helloworld => ./examples/protobuf/helloworld
-	github.com/firecracker-microvm/firecracker-containerd => github.com/ease-lab/firecracker-containerd v0.0.0-20200804113524-bc259c9e8152
+	github.com/firecracker-microvm/firecracker-containerd => github.com/ease-lab/firecracker-containerd v0.0.0-20210529101248-e628ce108f12
 	github.com/firecracker-microvm/firecracker-go-sdk => github.com/ease-lab/firecracker-go-sdk v0.20.1-0.20200625102438-8edf287b0123
```
[the remark this answers is not in the document]
Comment (plamenmpetrov) [reply go.mod:45]: This dependency is no longer required.

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (ustiugov) [reply go.mod:6]: Yes please

**Comment by ustiugov** [comment]
Rebase when ready and let's merge it

File: scripts/clean_fcctr.sh [hunk 3]
```diff
@@ -69,7 +69,8 @@ for d in containerd shim-base snapshotter; do sudo rm -rf /var/lib/firecracker-c
 echo Cleaning /run/firecracker-containerd/*
 sudo rm -rf /run/firecracker-containerd/containerd.sock.ttrpc \
     /run/firecracker-containerd/io.containerd.runtime.v1.linux \
-    /run/firecracker-containerd/io.containerd.runtime.v2.task
+    /run/firecracker-containerd/io.containerd.runtime.v2.task \
+    /run/containerd/s
```
[the remark this answers is not in the document]
Comment (plamenmpetrov) [reply scripts/clean_fcctr.sh:73]: Not incomplete. This deletes any remaining shim sockets from containerd. https://github.com/containerd/containerd/blob/36cc874494a56a253cd181a1a685b44b58a2e34a/runtime/v2/shim/util_unix.go#L74

I remember having an issue where the shim socket was not cleaned up.

File: ctriface/go.mod [hunk 4]
```diff
@@ -2,11 +2,16 @@ module github.com/ease-lab/vhive/ctriface
 
 go 1.15
 
-// Workaround for github.com/containerd/containerd issue #3031
-replace github.com/docker/distribution v2.7.1+incompatible => github.com/docker/distribution v2.7.1-0.20190205005809-0d3efadf0154+incompatible
+replace (
+	// Pin gPRC-related dependencies as like containerd v1.5.1
```
[the remark this answers is not in the document]
Comment (plamenmpetrov) [reply ctriface/go.mod]: I copied this replace block from firecracker-containerd.

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (ustiugov) [reply go.mod:45]: then need to remove it

**Comment by ustiugov** [comment]
please add a note to CHANGELOG.md in the Changed section of unreleased

**Review by ustiugov — APPROVED** [review]

**Pull request merged**